fappdeploy/httptrigger/GHubEventDispatch.cs

```diff
@@ -1,130 +1,213 @@
+        return requestObject[0]["data"];
+    }
+    private static dynamic ParseMapsEvent(dynamic requestObject)
+    {
+        return requestObject[0]["data"];
+    }
+    private static dynamic ParseMediaEvent(dynamic requestObject)
+    {
+        return requestObject[0]["data"];
+    }
+    private static dynamic ParseResourcesEvent(dynamic requestObject)
+    {
+        return requestObject[0]["data"];
+    }
+    private static dynamic ParseServiceBusEvent(dynamic requestObject)
+    {
+        return requestObject[0]["data"];
+    }
+    private static dynamic ParseSignalRServiceEvent(dynamic requestObject)
+    {
+        return requestObject[0]["data"];
+    }
 
-public static class GridEventHandler{
-    [FunctionName("PrettyPoisons")]
-    public static async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "post", Route = null)]HttpRequestMessage req, ILogger log, ExecutionContext context)
+    [FunctionName("generic_triggers")]
+    public static async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequestMessage req, ILogger log, ExecutionContext context)
     {
         log.LogInformation("C# HTTP trigger function processed a request.");
 
         string requestBody = await req.Content.ReadAsStringAsync();
         dynamic requestObject = JsonConvert.DeserializeObject(requestBody);
-        var webhook_res = string.Empty;
         var current_event = requestObject[0]["eventType"].ToString();
 
-        if (current_event == "Microsoft.EventGrid.SubscriptionValidationEvent" ){
-            if (requestObject != null && requestObject[0]["data"] != null){
-                var validationCode = requestObject[0].data.validationCode;
-                if(validationCode != null){
-                webhook_res= Newtonsoft.Json.JsonConvert.SerializeObject(new Newtonsoft.Json.Linq.JObject {["validationResponse"]= validationCode});
+        if (current_event == "Microsoft.EventGrid.SubscriptionValidationEvent")
+        {
+            string webhook_res = ParseEventGridValidationCode(requestObject);
+            if (!string.IsNullOrEmpty(webhook_res))
+            {
                 return (ActionResult)new OkObjectResult($"{webhook_res}");
-                }
             }
         }
-        
 
-        if (current_event.Contains("Microsoft.MachineLearningServices"))
+        string[] event_data = current_event.Split(".");
+        string event_source = string.Empty;
+        string event_type = string.Empty;
+
+        if (event_data.Length > 1)
+        {
+            event_source = event_data[1];
+        }
+
+        var queryParams = System.Web.HttpUtility.ParseQueryString(req.RequestUri.Query);
+        string repo_name = queryParams.Get("repoName");
+        log.LogInformation("fetching repo name from query parameters." + repo_name);
+
+        if (event_data.Length > 2 && !string.IsNullOrEmpty(repo_name))
         {
+            event_type = event_source.ToLower()  ;
+
```

Comment:
what exactly this part is doing?

Reply:
if event triggered is Microsoft.KeyVault.KeyExpired, then
event_source will be KeyVault and 
event_type will be keyvault-keyexpired.
basically we are appending the event_source as prefix in the event_type.